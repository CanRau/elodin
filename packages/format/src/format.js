import { parse } from '@elodin/parser'

import formatFromAST from './formatFromAST'

export default function format(input, config = {}) {
  const parsed = parse(input)

  if (parsed.errors.length === 0) {
    return formatFromAST(parsed.ast, config)
  }

  return input
}
