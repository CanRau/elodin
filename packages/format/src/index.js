import format from './format'
import formatFromAST from './formatFromAST'

export { format, formatFromAST }
