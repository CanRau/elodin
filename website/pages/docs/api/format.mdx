import { make as DocLayout } from '../../../components/DocLayout.bs.js'

export default DocLayout

# Format

`@elodin/format`

Format beautifies a string of Elodin code in an opinionated approach.

## format

It exposes a single `format` method that just takes an single string of code.

### Example

```javascript
import { format } from '@elodin/format'

const input = `
view Button { backgroundColor: red
      paddingLeft: 10
paddingRight  :   10
}
`

const output = format(input)

output = `
view Button {
  backgroundColor: red
  paddingLeft: 10
  paddingRight: 10
}
`
```
