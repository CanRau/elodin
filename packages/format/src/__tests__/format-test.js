import format from '../format'

const file = `
fragment Flex {
   flexGrow: 1
  flexDirection: row
}
view Button{justifyContent: flexStart


      __borderWidth: .3
       backgroundColor: rgb(  255  200   155  )

  [ Type =   Primary  ] {  paddingLeft: 20[Mode=Dark]{ backgroundColor:green}}
  }

   text    Label {
lineHeight:   $lineHeight
    fontSize: 20

     [@hover] {
       color:red
     }

     [@minWidth=320] {
       color:blue [@hover] {
         color: green
       }
     }
}`

describe('Formatting elodin syntax', () => {
  it('should return beautiful syntax', () => {
    expect(format(file).code).toMatchSnapshot()
  })

  it('should return beautiful syntax with custom syntax', () => {
    expect(
      format(
        file,
        {
          ident: '    ',
        }.code
      )
    ).toMatchSnapshot()
  })
})
