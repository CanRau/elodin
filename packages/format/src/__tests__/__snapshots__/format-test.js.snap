// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formatting elodin syntax should return beautiful syntax 1`] = `
"variant Mode {
  Dark
  Light
}

fragment Flex {
  flexGrow: 1
  flexDirection: row
}

view Button {
  justifyContent: flexStart
  __borderWidth: .3
  backgroundColor: rgb(255 200 155)
  [Type=Primary] {
    paddingLeft: 20
    [Mode=Dark] {
      backgroundColor: rgb(0 128 0)
    }
  }
}

text Label {
  lineHeight: $lineHeight
  fontSize: 20
  [@hover] {
    color: rgb(255 0 0)
  }
  [@minWidth=320] {
    color: rgb(0 0 255)
    [@hover] {
      color: rgb(0 128 0)
    }
  }
}"
`;

exports[`Formatting elodin syntax should return beautiful syntax with custom syntax 1`] = `
Object {
  "code": "variant Mode {
  Dark
  Light
}

fragment Flex {
  flexGrow: 1
  flexDirection: row
}

view Button {
  justifyContent: flexStart
  __borderWidth: .3
  backgroundColor: rgb(255 200 155)
  [Type=Primary] {
    paddingLeft: 20
    [Mode=Dark] {
      backgroundColor: rgb(0 128 0)
    }
  }
}

text Label {
  lineHeight: $lineHeight
  fontSize: 20
  [@hover] {
    color: rgb(255 0 0)
  }
  [@minWidth=320] {
    color: rgb(0 0 255)
    [@hover] {
      color: rgb(0 128 0)
    }
  }
}",
  "errors": Array [],
}
`;
